src/worker/tasks/updateMinedTx.ts

```diff
@@ -59,14 +60,19 @@ export const updateMinedTx = async () => {
                 )) as ethers.providers.TransactionResponse | null;
 
               // Get the timestamp when the transaction was mined
-              const minedAt = new Date(
-                (
-                  await getBlock({
-                    block: receipt.blockNumber,
-                    network: sdk.getProvider(),
-                  })
-                ).timestamp * 1000,
-              );
+              const block = await getBlock({
+                block: receipt.blockNumber,
+                network: sdk.getProvider(),
+              });
+              if (!block) {
+                throw `Block not found. txHash=${
```

Comment:
Logging this until we find a repro case where the block returns without a valid timestamp.